### PR TITLE
Refactoring of deployment pages

### PR DESCRIPTION
**Pull request by honza-kasik**
* Both deployment pages are now one - `DeploymentsPage`
* `@Location` moved to separate classes, respectively `DomainDeploymentEntryPoint` and `StandaloneDeploymentEntryPoint`
* Refactored
* Moved deployment related method from `org.jboss.hal.testsuite.test.runtime.SimpleWebserviceEndpointTestCase` to `DeploymentsPage`
* Added Javadoc

### CODE REVIEW

File: common/src/main/java/org/jboss/hal/testsuite/page/runtime/DeploymentsPage.java [hunk 1]
```diff
@@ -2,36 +2,43 @@
...
 /**
- * @author jcechace
+ * Abstraction over 'Deployments' page for both domain and standalone mode.
  */
-@Location("#domain-deployments")
-public class DomainDeploymentPage extends ConfigurationPage {
+public class DeploymentsPage extends ConfigurationPage {
 
     private static final By BACK_BUTTON = By.xpath(".//a[text()='Back']");
     private static final By CONTENT = By.className(PropUtils.get("page.content.gwt-layoutpanel"));
```
Comment (zebrik) [inline common/src/main/java/org/jboss/hal/testsuite/page/runtime/DeploymentsPage.java:32]: Please define and use private method for check actual standalone/domain mode and fast fail if needed.

**Review by zebrik — COMMENTED** [review]
Just one tiny requirement for make code more error resistent.

**Comment by zebrik** [comment]
@honza-kasik please resolve conflicts caused by https://github.com/hal/testsuite/pull/207 merge.

**Comment by honza-kasik** [comment]
@zebrik Conflicts resolved. Ready to be merged!

**Comment by zebrik** [comment]
Please squash and I will merge this.

**Comment by honza-kasik** [comment]
Thanks, squashed and ready to be merged.

**Comment by zebrik** [comment]
Merging. Thank you.

**Pull request merged**